test/tokenize.jl

```diff
@@ -307,218 +325,259 @@ end
     @test tok("0b0101").kind==K"BinInt"
 end
 
-@testset "show" begin
-    io = IOBuffer()
-    show(io, collect(tokenize("\"abc\nd\"ef"))[2])
-    @test String(take!(io)) == "1-5        String         "
+function check_lexing(str, results::Vector{Pair{Kind, String}})
+    offset = 0
+    ts = collect(tokenize(str))
+    for (tok, res) in zip(ts, results)
+        @test tok.kind == res[1]
+        @test untokenize(tok, str, offset) == res[2]
+        offset = tok.endbyte+1
+    end
 end
 
-~(tok::Token, t::Tuple) = tok.kind == t[1] && untokenize(tok, t[3]) == t[2]
-
 @testset "raw strings" begin
     str = raw""" str"x $ \ y" """
-    ts = collect(tokenize(str))
-    @test ts[1] ~ (K"Whitespace" , " "        , str)
-    @test ts[2] ~ (K"Identifier" , "str"      , str)
-    @test ts[3] ~ (K"\""         , "\""       , str)
-    @test ts[4] ~ (K"String"     , "x \$ \\ y", str)
-    @test ts[5] ~ (K"\""         , "\""       , str)
-    @test ts[6] ~ (K"Whitespace" , " "        , str)
-    @test ts[7] ~ (K"EndMarker"  , ""         , str)
+
+    check_lexing(str,
+    [
+        K"Whitespace" => " "        ,
+        K"Identifier" => "str"      ,
+        K"\""         => "\""       ,
+        K"String"     => "x \$ \\ y",
+        K"\""         => "\""       ,
+        K"Whitespace" => " "        ,
+        K"EndMarker"  => ""         ,
+    ])
```

Comment:
This really looks a lot nicer :+1:

I guess the only downside of this is we don't get so much useful line information when a test fails as the `@test` macro occurs inside `check_lexing()`. A possible alternative could be to rearrange like this

```julia
@test lex(str) ==  [
        K"Whitespace" => " "        ,
        K"Identifier" => "str"      ,
...
]
```